src/MongoDB.EntityFrameworkCore/Infrastructure/MongoModelValidator.cs

```diff
@@ -72,7 +72,8 @@ public override void Validate(IModel model, IDiagnosticsLogger<DbLoggerCategory.
         ValidateMaximumOneRowVersionPerEntity(model);
         ValidateNoUnsupportedAttributesOrAnnotations(model);
         ValidateElementNames(model);
-        ValidateNoShadowProperties(model);
+        ValidateNoUnsupportedShadowProperties(model);
+        ValidateElementNames(model);
```

Comment:
You have `ValidateElementsNames` some row earlier too, is it supposed to be like this?

Reply:
No, looks like something that came out around the rebasing. I'll fix.